packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -1647,6 +1652,187 @@ function cleanupParameterData(inputData: NodeParameterValueType): NodeParameterV
 	return inputData;
 }
 
+function findPropertyInArray(
+	name: string,
+	options: INodePropertyMode[],
+): INodePropertyMode | undefined;
+function findPropertyInArray(
+	name: string,
+	options: Array<INodePropertyOptions | INodeProperties | INodePropertyCollection>,
+): INodePropertyOptions | INodeProperties | INodePropertyCollection | undefined;
+function findPropertyInArray(
+	name: string,
+	options: Array<
+		INodePropertyOptions | INodeProperties | INodePropertyCollection | INodePropertyMode
+	>,
+):
+	| INodePropertyOptions
+	| INodeProperties
+	| INodePropertyCollection
+	| INodePropertyMode
+	| undefined {
+	const prop = options.find((i) => i.name === name);
+	return prop;
+}
+
+function findPropertyFromParameterName(
+	parameterName: string,
+	nodeType: INodeType,
+): INodePropertyOptions | INodeProperties | INodePropertyCollection {
+	let property: INodePropertyOptions | INodeProperties | INodePropertyCollection | undefined;
+	const paramParts = parameterName.split('.');
+
+	property = findPropertyInArray(paramParts.shift()!, nodeType.description.properties);
+	if (!property) {
+		throw new Error(`Couldn't not find property "${parameterName}"`);
+	}
+
+	// eslint-disable-next-line no-restricted-syntax
+	for (const p of paramParts) {
+		const param = p.split('[')[0];
+		if ('options' in property && property.options) {
+			property = findPropertyInArray(param, property.options);
+		} else if ('values' in property) {
+			property = findPropertyInArray(param, property.values);
+		} else {
+			throw new Error(`Couldn't not find property "${parameterName}"`);
+		}
+		if (!property) {
+			throw new Error(`Couldn't not find property "${parameterName}"`);
+		}
+	}
+
+	return property;
+}
+
+function executeRegexExtractValue(
+	value: string,
+	regex: RegExp,
+	parameterName: string,
+): NodeParameterValueType | object {
+	const extracted = regex.exec(value);
+	if (!extracted) {
+		throw new Error(
+			`extractValue for "${parameterName}" could not extract value. This likely means that the supplied value doesn't match the extractor regex.`,
+		);
+	}
+	if (extracted.length < 2 || extracted.length > 2) {
+		throw new Error(
+			`Property "${parameterName}" has an invalid extractValue regex "${regex.source}". extractValue expects exactly one group to be returned.`,
+		);
+	}
+	return extracted[1];
+}
+
+function extractValueRLC(
```

Comment:
`extractValueRLC` and `extractValueOther` are largely similar to each other. Does it make sense for them to be separate? If they were consolidated, `executeRegexExtractValue` wouldn't be needed twice, so it could also be brought in, as it's only a thin wrapper for the regex call and two guards. If we keep them separate, I'd reconsider the functions naming.

Reply:
I originally did have them combined but it was a lot harder to follow due to lots of type assertions and other `if/else`s that because of the slight differences. I decided to refactor it out into 2 functions for readabilities sake.